### DISCUSSION TITLE
broken thumbnail & picture in web app for newly uploaded photos

### DISCUSSION THREAD

**Discussion by fuhaoda** [immich-app/immich#11409] [General]
After I upgrade it to v1.110.0, and upload new images from my phone.  I can't see the thumbnail and the pictures from web app. The old files are OK. Here is the image to illustrate this issue, with my yml file and .env file.
<img width="1578" alt="Screenshot 2024-07-27 at 10 19 53 PM" src="https://github.com/user-attachments/assets/ed66bb28-48e1-4c7e-b65f-08a5af662684">

My yml file.
```
#
# WARNING: Make sure to use the docker-compose.yml of the current release:
#
# https://github.com/immich-app/immich/releases/latest/download/docker-compose.yml
#
# The compose file on main may not be compatible with the latest release.
#

name: immich

services:
  immich-server:
    container_name: immich_server
    image: ghcr.io/immich-app/immich-server:${IMMICH_VERSION:-release}
    # extends:
    #   file: hwaccel.transcoding.yml
    #   service: cpu # set to one of [nvenc, quicksync, rkmpp, vaapi, vaapi-wsl] for accelerated transcoding
    volumes:
      - ${UPLOAD_LOCATION}:/usr/src/app/upload
      - /etc/localtime:/etc/localtime:ro
    env_file:
      - .env
    ports:
      - 2283:3001
    depends_on:
      - redis
      - database
    restart: always

  immich-machine-learning:
    container_name: immich_machine_learning
    # For hardware acceleration, add one of -[armnn, cuda, openvino] to the image tag.
    # Example tag: ${IMMICH_VERSION:-release}-cuda
    image: ghcr.io/immich-app/immich-machine-learning:${IMMICH_VERSION:-release}
    # extends: # uncomment this section for hardware acceleration - see https://immich.app/docs/features/ml-hardware-acceleration
    #   file: hwaccel.ml.yml
    #   service: cpu # set to one of [armnn, cuda, openvino, openvino-wsl] for accelerated inference - use the `-wsl` version for WSL2 where applicable
    volumes:
      - model-cache:/cache
    env_file:
      - .env
    restart: always

  redis:
    container_name: immich_redis
    image: docker.io/redis:6.2-alpine@sha256:328fe6a5822256d065debb36617a8169dbfbd77b797c525288e465f56c1d392b
    healthcheck:
      test: redis-cli ping || exit 1
    restart: always

  database:
    container_name: immich_postgres
    image: docker.io/tensorchord/pgvecto-rs:pg14-v0.2.0@sha256:90724186f0a3517cf6914295b5ab410db9ce23190a2d9d0b9dd6463e3fa298f0
    environment:
      POSTGRES_PASSWORD: ${DB_PASSWORD}
      POSTGRES_USER: ${DB_USERNAME}
      POSTGRES_DB: ${DB_DATABASE_NAME}
      POSTGRES_INITDB_ARGS: '--data-checksums'
    volumes:
      - pgdata:/var/lib/postgresql/data
# - ${DB_DATA_LOCATION}:/var/lib/postgresql/data
    healthcheck:
      test: pg_isready --dbname='${DB_DATABASE_NAME}' --username='${DB_USERNAME}' || exit 1; Chksum="$$(psql --dbname='${DB_DATABASE_NAME}' --username='${DB_USERNAME}' --tuples-only --no-align --command='SELECT COALESCE(SUM(checksum_failures), 0) FROM pg_stat_database')"; echo "checksum failure count is $$Chksum"; [ "$$Chksum" = '0' ] || exit 1
      interval: 5m
      start_interval: 30s
      start_period: 5m
    command: ["postgres", "-c" ,"shared_preload_libraries=vectors.so", "-c", 'search_path="$$user", public, vectors', "-c", "logging_collector=on", "-c", "max_wal_size=2GB", "-c", "shared_buffers=512MB", "-c", "wal_compression=on"]
    restart: always

volumes:
  pgdata:
  model-cache:
```

my `.env` file
```
# You can find documentation for all the supported env variables at https://immich.app/docs/install/environment-variables

# The location where your uploaded files are stored
UPLOAD_LOCATION=/mnt/share/public/family_memory/immich

# The Immich version to use. You can pin this to a specific version like "v1.71.0"
IMMICH_VERSION=release

# Connection secret for postgres. You should change it to a random password
DB_PASSWORD=abcdefg

# The values below this line do not need to be changed
###################################################################################
DB_HOSTNAME=immich_postgres
DB_USERNAME=postgres
DB_DATABASE_NAME=immich

REDIS_HOSTNAME=immich_redis
```

**Comment by alextran1502** [comment]
Probably because you have this in your .env file

```
REDIS_HOSTNAME=immich_redis```
```

instead of

```
REDIS_HOSTNAME=immich_redis
```

**Reply by fuhaoda** [reply]
I use 
```
REDIS_HOSTNAME=immich_redis
```
Just updated my question. It still doesn't work. Anything else I should try? I saw a similar issue reported here [click](https://www.reddit.com/r/immich/s/Z77JOgl39e)

**Reply by fuhaoda** [reply]
I used the old way to put   - 
`pgdata:/var/lib/postgresql/data`

does this cause the issue?

**Reply by mertalev** [reply]
No, using `pgdata` should be fine.

Do you see any errors in the server logs? You can try clicking one of the broken images and choosing the "Refresh Thumbnails" option. If it fails, you'll see an error in the logs.

**Reply by fuhaoda** [reply]
@mertalev @alextran1502 
Following your suggestion, I "refresh thumbnails", and use 
`sudo docker logs <immich-server:release container ID>`
I got the following. It didn't report any error, but I still can't see the preview.
I can see the images EXIF information, from the Web App. I have tried multiple computer, browsers to rule out other factors. 


```
[Nest] 17  - 07/28/2024, 12:59:04 PM     LOG [Api:AuditService~3rf9y8xs] Found 94028 assets, 6 users, 1000 people
[Nest] 17  - 07/28/2024, 12:59:04 PM     LOG [Api:AuditService~3rf9y8xs] Found 94028 assets, 6 users, 1000 people
[Nest] 17  - 07/28/2024, 12:59:04 PM     LOG [Api:AuditService~3rf9y8xs] Found 94028 assets, 6 users, 470 people
[Nest] 7  - 07/28/2024, 1:12:45 PM     LOG [Microservices:MediaService] Successfully generated JPEG image preview for asset 77dcc05e-5679-4578-a09e-c8ee706caf1e
```

BTW, my upload location is in a `/mnt/` folder.  I didn't have any issue for previous version. Will that cause a problem in the new version?
`UPLOAD_LOCATION=/mnt/share/public/family_memory/immich`

**Reply by mertalev** [reply]
It's very strange that the image gets processed, but there's still no thumbnail or preview for it in the web app. Is there an example image you could share that has this issue (in a .zip file so GitHub doesn't change it)?

**Reply by fuhaoda** [reply]
[IMG_1138.HEIC.zip](https://github.com/user-attachments/files/16404528/IMG_1138.HEIC.zip)
They are standard iPhone pictures in HEIC format. Before the immich update, I have been using it for a while. See the attachment for an example. @mertalev

**Reply by mertalev** [reply]
I see the same issue with this image. Strangely enough, metadata extraction also doesn't find anything. Every DB column in the `exif` is null except for `dateTimeOriginal` and `modifyDate` - even the file size is 0. But using the `exiftool` CLI shows normal metadata.

What was the Immich version you used before upgrading to 1.110.0?

**Reply by fuhaoda** [reply]
I can't remember exactly. Maybe 1.098? It is about 4 months ago.  Not only for this image, but for all the others using my phone and my wife's phone.

**Comment by fuhaoda** [comment]
I just found that if I use my phone to regenerate the thumbnails, the problem can be fixed on Web App. It seems that this could be a bug when using phone to upload the pictures and the thumbnails and images can't be correctly displayed in Web App.

**Comment by byjlw** [comment]
I'm having this same issue right now. I didn't upgrade but i spun it up for the first time. 
If i upload via the web interface it's fine but the all the images it picked up via external-library are like this. 
Oddly videos show the thumbnail and will play correctly.

**Reply by alextran1502** [reply]
@byjlw do you use the default setup or you modified yours? Can you supply your compose and .env file?

**Comment by StephenDaDev** [comment]
I came across this last night when uploading from a Samsung S21 Ultra. I found that if I selected all of the broken photos (I cheated and used CMD/CTRL+A as there weren't that many in the account not from that phone) and pressed regenerate thumbnails it solved the problem. I didn't have to do it from the mobile app as noted above, using the web app worked fine for me. Same version `v1.110.0`

**Reply by alextran1502** [reply]
Are you using the docker-compose setup as well?

**Reply by StephenDaDev** [reply]
Yes, unmodified.

**Reply by alextran1502** [reply]
Can you try docker compose down, then docker compose pull and finally docker compose up?

**Reply by StephenDaDev** [reply]
Done. Looking through I don't see any that are still screwed up but I'm not sure if I got them all with the regenerate thumbnail workaround earlier. I will take some more and upload them and get back on that.

**Reply by StephenDaDev** [reply]
Have not updated yet, still on v1.110.0. It appears to have solved the issue with new uploads. I will upgrade right now and test with a few more uploads.

**Reply by StephenDaDev** [reply]
No issue v1.111.0
